pkg/download/consistent_hashing.go

```diff
@@ -262,7 +261,28 @@ func (m *ConsistentHashingMode) DoRequest(ctx context.Context, start, end int64,
 
 	resp, err := m.Client.Do(req)
 	if err != nil {
-		return nil, fmt.Errorf("error executing request for %s: %w", req.URL.String(), err)
+		if errors.Is(err, client.ErrStrategyFallback) {
+			origErr := err
+			req, err := http.NewRequestWithContext(chContext, "GET", urlString, nil)
+			if err != nil {
+				return nil, fmt.Errorf("failed to download %s: %w", req.URL.String(), err)
+			}
+			_, err = m.rewriteRequestToCacheHost(req, start, end, cachePodIndex)
+			if err != nil {
+				// return origErr so that we can use our regular fallback strategy
+				return nil, origErr
+			}
+			req.Header.Set("Range", fmt.Sprintf("bytes=%d-%d", start, end))
+			logger.Debug().Str("url", urlString).Str("munged_url", req.URL.String()).Str("host", req.Host).Int64("start", start).Int64("end", end).Msg("retry request")
+
+			resp, err = m.Client.Do(req)
+			if err != nil {
+				// return origErr so that we can use our regular fallback strategy
+				return nil, origErr
```

Comment:
I think we're going to want this for the foreseeable future. As reliability is the real goal, I want to be durable for "cache disappears" but we still serve inference.

That said, I hear the earlier comments about "do we need this".